File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Comment/SelectCommentDTO.cs
using System;
using System.Collections.Generic;
using platform_core_service.Common.Entities.BaseEntity;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Comment
{
    public class SelectCommentDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;

        public string Content { get; set; } = null!;

        public string AuthorId { get; set; } = null!;

        public string? PostId { get; set; }

        public string? AnswerId { get; set; }

        public string? ReplyToCommentId { get; set; }

        public int UpvoteCount { get; set; } = 0;

        public int DownvoteCount { get; set; } = 0;

        public DateTimeOffset DateCreated { get; set; }

        public DateTimeOffset? DateModified { get; set; }

        public SelectCommentAuthorDTO? Author { get; set; }

        public List<SelectCommentDTO> Replies { get; set; } = new List<SelectCommentDTO>();
    }

    public class SelectCommentAuthorDTO
    {
        public string Id { get; set; } = null!;

        public string FirstName { get; set; } = null!;

        public string LastName { get; set; } = null!;
    }
}

File: frontend/src/types/comment/select-comment-dto.ts
export interface SelectCommentAuthorDTO {
    id: string;
    fullName: string;
    avatarUrl?: string;
    backgroundUrl?: string;
    bio: string;
    reputationPoints: number;
    techStacks: string[];
    isPrivate: boolean;
}

export interface SelectCommentDTO {
    id: string;
    content: string;
    authorId: string;
    postId?: string;
    answerId?: string;
    replyToCommentId?: string;
    upvoteCount: number;
    downvoteCount: number;
    dateCreated: string; // Khuyên dùng string (ISO 8601) cho DateTimeOffset
    dateModified?: string;
    author?: SelectCommentAuthorDTO;

    currentUserVote?: boolean | null; // null = no vote, true = upvoted, false = downvoted

    // Thuộc tính đệ quy: Một comment có thể chứa nhiều comment con (replies)
    replies: SelectCommentDTO[];
}